src/modules/main/PlayerView.tsx

```diff
@@ -1,23 +1,169 @@
+      }));
+
+      setAnsweredLabels(answered);
+      setLabels(settingLabels);
+    }
+  };
+
+  const submit = (): void => {
+    const submittedAnswers = answeredLabels.map(({ expected, actual }) => ({
+      expectedId: expected.id,
+      actualId: actual?.id,
+    }));
+    saveAppData({
+      data: { answers: submittedAnswers },
+      type: ANSWERS_SUBMISSION_TYPE,
+    });
+  };
+
+  const lastSubmittedAnsweredLabels = useMemo(
+    () =>
+      answers?.map(({ expectedId, actualId }: SubmittedAnswer) => ({
+        expected: settingLabels?.find(({ id }) => id === expectedId) as Label,
+        actual: settingLabels?.find(({ id }) => id === actualId) || null,
+      })),
+    [answers, settingLabels],
+  );
+
+  useEffect(() => {
+    if (!settingLabels) {
+      return;
+    }
+    if (lastSubmittedAnsweredLabels) {
+      setAnsweredLabels(lastSubmittedAnsweredLabels);
+      setLabels(
+        settingLabels.filter(
+          ({ id }) =>
+            !lastSubmittedAnsweredLabels.find(
+              ({ actual }) => actual?.id === id,
+            ),
+        ),
+      );
+    } else {
+      const answered = settingLabels.map((label) => ({
+        expected: label,
+        actual: null,
+      }));
+
+      setAnsweredLabels(answered);
+      setLabels(settingLabels);
+    }
+  }, [lastSubmittedAnsweredLabels, settingLabels]);
+
+  const isSubmitted = useMemo(
+    () => isEqual(answeredLabels, lastSubmittedAnsweredLabels),
+    [answeredLabels, lastSubmittedAnsweredLabels],
+  );
+
+  if (isLoading) {
+    return <CircularProgress />;
+  }
+
+  if (!image || !settingLabels) {
+    return <Alert severity="error">{t(APP.UNCONFIGURED_ITEM)}</Alert>;
+  }
+
+  const onLabelMoved = (
+    newLabels: Label[],
+    newAnswers: AnsweredLabel[],
+  ): void => {
+    setLabels(newLabels);
+    setAnsweredLabels(newAnswers);
+  };
 
   return (
-    <div data-cy={PLAYER_VIEW_CY}>
-      Player as {permission}
-      <Box p={2}>
-        <Typography>Members</Typography>
-        <pre>{JSON.stringify(members, null, 2)}</pre>
-      </Box>
-    </div>
+    <Container data-cy={PLAYER_VIEW_CY}>
+      <Stack spacing={2} padding={2}>
+        <Stack justifyContent="space-between" flexDirection="row">
+          <Box>
+            <Typography variant="h5" fontWeight="bold">
+              {appContext?.item.name}
+            </Typography>
+            {appSettings?.[0]?.data.description && (
+              <Typography variant="body1">
+                {appSettings?.[0]?.data.description}
+              </Typography>
+            )}
+          </Box>
+          {isSubmitted ? (
+            answersAppData &&
+            answersAppData.length > 0 && (
+              <Button onClick={retry}>{t(APP.RETRY)}</Button>
+            )
+          ) : (
+            <Button
+              onClick={submit}
+              variant="contained"
+              sx={{ height: 'fit-content' }}
+            >
+              {t(APP.SUBMIT)}
+            </Button>
```

Comment:
Actually I think it doesn't have to be one or the other. Why not let the user "retry/clean" the labels without having to submit?

Reply:
We had a discussion with @spaenleh and @Defferrard , and we thought that we don't need both together.
In case of `clean` I think it's feasible to have it anyway.